src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -738,6 +738,12 @@ AdaptiveExecutor(CitusScanState *scanState)
 	/* we should only call this once before the scan finished */
 	Assert(!scanState->finishedRemoteScan);
 
+	MemoryContext localContext = AllocSetContextCreate(CurrentMemoryContext,
```

Comment:
Yes, it's only for debugging purposes, including MemoryContext dumps that happen on OOM in production. We should probably have more of them for high-level, memory-intensive code blocks.

Introducing early returns in these 2 functions would lead to various other problems (e.g. what if you return before FinishDistributedExecution), so I'm not too concerned about that.